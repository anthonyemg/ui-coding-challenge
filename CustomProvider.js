import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './src/public/reducers';

const store = createStore(reducer);

const CustomProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

export default CustomProvider;
