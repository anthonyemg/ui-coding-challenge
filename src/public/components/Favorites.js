import React from 'react';
import { connect } from 'react-redux';
import { updateFavorites, toggleFavoritesModal } from '../actions';
import Modal from './Modal';
import FavoritesModal from './FavoritesModal';

class Favorites extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      favorites: [...this.props.favorites]
    };
    this.handleToggleFavoritesModal = this.handleToggleFavoritesModal.bind(this);
  }
  handleToggleFavoritesModal() {
    this.props.toggleFavoritesModal(!this.props.displayFavoritesModal);
  }
  render() {
    return (
      <div className="component-container">
        <div className="component-wrapper">
          <div className="component-details">
            <span className="component-title">Favorite Motorcycles</span>

            {this.props.favorites.length === 0 ? (
              <span className="component-noneAdded">None Added</span>
            ) : (
              this.props.favorites.map((favorite, index) => (
                <span className="component-text" key={index}>
                  {index + 1}. {favorite}
                </span>
              ))
            )}
          </div>
          <div>
            <button
              className="component-editButton"
              onClick={() => this.handleToggleFavoritesModal()}
            >
              Edit Motorcyles
            </button>
          </div>
        </div>
        {this.props.displayFavoritesModal && (
          <Modal>
            <FavoritesModal
              favorites={this.props.favorites}
              updateFavorites={this.props.updateFavorites}
              handleToggleFavoritesModal={this.handleToggleFavoritesModal}
            />
          </Modal>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  favorites: state.favorites,
  displayFavoritesModal: state.displayFavoritesModal
});
const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch(updateFavorites(favorites)),
  toggleFavoritesModal: boolean => dispatch(toggleFavoritesModal(boolean))
});

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
