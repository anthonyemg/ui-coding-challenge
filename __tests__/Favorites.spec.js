import React from 'react';
import renderer from 'react-test-renderer';
import CustomProvider from '../CustomProvider';
import Favorites from '../src/public/components/Favorites';

test('the Favorites component renders correctly', () => {
  const component = renderer.create(
    <CustomProvider>
      <Favorites />
    </CustomProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
