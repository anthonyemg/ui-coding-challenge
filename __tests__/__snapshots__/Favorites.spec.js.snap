// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Favorites component renders correctly 1`] = `
<div
  className="component-container"
>
  <div
    className="component-wrapper"
  >
    <div
      className="component-details"
    >
      <span
        className="component-title"
      >
        Favorite Motorcycles
      </span>
      <span
        className="component-noneAdded"
      >
        None Added
      </span>
    </div>
    <div>
      <button
        className="component-editButton"
        onClick={[Function]}
      >
        Edit Motorcyles
      </button>
    </div>
  </div>
</div>
`;
